src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(ShopStorage storage, FruitTransaction transaction) {
+        storage.updateFruitQuantity(transaction.getFruit(), -transaction.getQuantity());
```

Comment:
Implement a check to prevent negative balances. If a purchase would result in a negative quantity, throw a `RuntimeException` to maintain data integrity.